### PR TITLE
Make Fabric a per RootView setting

### PR DESCRIPTION
**Pull request by acoates-ms**
Make Fabric a per RootView setting.  This is how core works, and enables apps to more incrementally opt into Fabric in more complex brownfield apps.

This also fixes some issues where in playground you'd hit crashes switching back and forth between fabric and paper.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7843)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/ReactHost/React.h [hunk 1]
```diff
@@ -127,6 +128,9 @@ struct ReactViewOptions {
 
   //! Set of initial component properties. It is a JSON string.
   folly::dynamic InitialProps;
+
+  //! Use Fabric for this ReactView
+  bool UseFabric;
```
Comment (NickGerleman) [inline vnext/Microsoft.ReactNative/ReactHost/React.h:133]:
```suggestion
  bool UseFabric {false};
```

File: vnext/Microsoft.ReactNative/ReactRootView.idl [hunk 2]
```diff
@@ -36,5 +36,8 @@ namespace Microsoft.ReactNative
 
     DOC_STRING("Reloads the current @ReactRootView UI components.")
     void ReloadView();
+
+    DOC_STRING("Use Fabric for this ReactRootView")
+    Boolean ExperimentalUseFabric { get; set; };
```
Comment (NickGerleman) [inline vnext/Microsoft.ReactNative/ReactRootView.idl:41]: Inevitably there will be people who try to enable this if they see it in their Intellisense😂. I guess it won't do anything without the Quirk setting and compile-time flag though, but maybe that could be confusing. Not a huge issue if it's marked as experimental though.

Maybe overkill but we could probably hide the property in a different interface that we can QI to if we don't want people to see this in autocomplete or whatever.

**Review by NickGerleman — APPROVED** [review]

File: vnext/Microsoft.ReactNative/ReactRootView.idl (hunk 2 above)
Comment (asklar) [reply vnext/Microsoft.ReactNative/ReactRootView.idl:41]: can you please mark these as [experimental]? If not allowed on properties, can you put these on an interface that's marked [experimental]?

File: packages/playground/windows/playground/MainPage.cpp [hunk 3]
```diff
@@ -51,6 +51,10 @@ void MainPage::OnLoadClick(
   }
   host.InstanceSettings().JavaScriptBundleFile(bundleFile);
 
+  winrt::Microsoft::ReactNative::QuirkSettings::SetEnableFabric(
+      host.InstanceSettings(), x_UseFabric().IsChecked().GetBoolean());
+  ReactRootView().ExperimentalUseFabric(x_UseFabric().IsChecked().GetBoolean());
+
```
Comment (asklar) [inline packages/playground/windows/playground/MainPage.cpp:57]: why is it necessary to set both? wouldn't it be enough to just set it in the react root view?

Comment (acoates-ms) [reply packages/playground/windows/playground/MainPage.cpp:57]: One is a setting on the instance, which causes the instance to create some internal infrastructure that is needed to be in the instance to load fabric root views.  In particular it creates the fabric ui manager.  I could just have that always be created if building with the USE_FABRIC flag. -- I'll look at removing this setting then.

**Pull request merged**